x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/NativePrivilegeStore.java

```diff
@@ -99,6 +100,16 @@ public class NativePrivilegeStore {
         Setting.Property.NodeScope
     );
 
+    /**
+     * Determines how long get privileges calls will wait for an available security index.
+     * The default value of 0 bypasses all waiting-related logic entirely.
+     */
+    private static final TimeValue SECURITY_INDEX_WAIT_TIMEOUT = TimeValue.parseTimeValue(
+        System.getProperty("es.security.security_index.wait_timeout", null),
+        TimeValue.ZERO,
+        "system property <es.security.security_index.wait_timeout>"
```

Comment:
nit: why "system property" ? (i read that as in a Java system property, i.e. -Dsomething=true)

Reply:
Just because I'm tying into a method that's used for settings parsing and the value here is passed into the exception text `"failed to parse setting [" + settingName + "] ...` 

However, `es.security.security_index.wait_timeout` is _not_ a setting so I tried to make the message less confusing by adding the `system property <>` prop bit.

It's admittedly neither here nor there but we are the only ones who should ever be setting this value and therefore the failure message _should_ be irrelevant in any case.